pkg/gather/clusterconfig/clusterconfig.go

```diff
@@ -36,24 +37,41 @@ var (
 )
 
 type Gatherer struct {
-	client     configv1client.ConfigV1Interface
-	coreClient corev1client.CoreV1Interface
+	client        configv1client.ConfigV1Interface
+	coreClient    corev1client.CoreV1Interface
+	metricsClient rest.Interface
 
 	lock        sync.Mutex
 	lastVersion *configv1.ClusterVersion
 }
 
-func New(client configv1client.ConfigV1Interface, coreClient corev1client.CoreV1Interface) *Gatherer {
+func New(client configv1client.ConfigV1Interface, coreClient corev1client.CoreV1Interface, metricsClient rest.Interface) *Gatherer {
 	return &Gatherer{
-		client:     client,
-		coreClient: coreClient,
+		client:        client,
+		coreClient:    coreClient,
+		metricsClient: metricsClient,
 	}
 }
 
 var reInvalidUIDCharacter = regexp.MustCompile(`[^a-z0-9\-]`)
 
 func (i *Gatherer) Gather(ctx context.Context, recorder record.Interface) error {
 	return record.Collect(ctx, recorder,
+		func() ([]record.Record, []error) {
+			if i.metricsClient == nil {
+				return nil, nil
+			}
+			data, err := i.metricsClient.Get().AbsPath("federate").
+				Param("match[]", "ALERTS").
```

Comment:
@smarterclayton i wish this is more visible (thinking like defaults.go?)

Reply:
Once we have a full list I agree it should be in its own package.

Reply:
Just out of curiosity. Why do we use the federate API instead of e.g `api/v1/query`?